File: app/src/main/java/com/datarockets/mnchkn/fragments/charts/ChartsInteractor.java
package com.datarockets.mnchkn.fragments.charts;

import com.datarockets.mnchkn.models.Player;

import java.util.ArrayList;

import lecho.lib.hellocharts.model.LineChartData;

public interface ChartsInteractor {

    interface OnChartLoadedListener {
        void showPlayers(ArrayList<Player> players);
    }

    LineChartData loadLineChartData(int type);
    void loadPlayers(int type, OnChartLoadedListener listener);

}

File: app/src/main/kotlin/com/datarockets/mnchkn/fragments/charts/ChartsInteractor.kt
package com.datarockets.mnchkn.fragments.charts

import com.datarockets.mnchkn.models.Player
import lecho.lib.hellocharts.model.LineChartData
import java.util.*

interface ChartsInteractor {

    interface OnChartLoadedListener {
        fun showPlayers(players: ArrayList<Player>)
    }

    fun loadLineChartData(type: Int): LineChartData
    fun loadPlayers(type: Int, listener: OnChartLoadedListener)

}
